routes/tasks.js

```diff
@@ -0,0 +1,196 @@
+const express = require('express');
+const app = require('../app');
+const router = express.Router();
+const ObjectID = require("mongodb").ObjectID;
+
+/* GET tasks listing. */
+router.get('/', function(req, res, next) {
+  req.app.locals.tasks.find().toArray()
+    .then(result => {
+      res.setHeader('Content-Type', 'application/json');
+      res.status(200).send(result);
+    })
+    .catch(error => {
+      console.error(error);
+    });
+});
+
+
+/* GET tasks by User */
+router.get('/:user_id', function(req, res, next) {
+  const user_id  = ObjectID(req.params.user_id);
+  req.app.locals.tasks.find({user_id}).toArray()
+  .then(result => {
+    res.setHeader('Content-Type', 'application/json');
+    res.status(200).send(result);
+  })
+  .catch(err => {
+    res.status(503).end();
+    console.error(err);
+  });
+});
+
+/* POST new task, returns new created document ID */
+router.post('/:user_id', function(req, res, next) {
+  const user_id = ObjectID(req.params.user_id); 
+  const { description, isPublic, dueDate } = req.body;
+  const newTask = { 
+    description, 
+    isPublic,
+    dueDate,
+    user_id,
+    givenClaps: [],
+    mangoTransactions: [],
+    mangosGiven: 0,
+    subTasks: [],
+    isDone: false,
+    // NOTE: maybe timestamp should be created on the client so it will be congruent with the user's timezone 
```

Comment:
Oooh good point. So the timestamp that Date.now() created is a unix timestamp
`A Number representing the milliseconds elapsed since the UNIX epoch.`
^ the return field from the docs

So as long as our frontend converts it to display the correct conversion based on the timezone, i think storing this universal timestamp will be the same whether its set from the frontend or backend. It might be off by a few secs though, since it's the timestamp when the API call is received and now when it's sent - idc though :D

Reply:
Ahhh interesting, yeah if it's only off by a few seconds it shouldn't matter. Better to have it initialized on the backend then. Good find!